### PR TITLE
Do not try to make API calls if token is not set for rfml lookups

### LINKED ISSUE #131: rainforest validate doesn't check if there is a token
When you execute `rainforest validate` without a token, HttpParty will through an error because there's a nil value in the headers:

```bash
/Users/juliiankrispel/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http/header.rb:18:in `block in initialize_http_header': undefined method `strip' for nil:NilClass (NoMethodError)
```

### PR DESCRIPTION
**Pull request by epaulet**
fixes #131

### CODE REVIEW

**Comment by epaulet** [comment]
fixes #129 as well

File: lib/rainforest/cli/remote_tests.rb [hunk 1]
```diff
@@ -13,29 +13,6 @@ def rfml_ids
     @rfml_ids ||= primary_key_dictionary.keys
   end
 
-  def tests
```
Comment (epaulet) [inline lib/rainforest/cli/remote_tests.rb]: this method doesn't seem to be used anymore. The endpoint is too slow anyway.

**Comment by epaulet** [comment]
Old behavior is back!
```
$ bin/rainforest validate
I, [2016-06-10T00:20:25.672454 #48419]  INFO -- : No test folder supplied. Using default folder: ./spec/rainforest
I, [2016-06-10T00:20:25.678040 #48419]  INFO -- : Validating parsing errors...
I, [2016-06-10T00:20:25.678097 #48419]  INFO -- : Validating embedded test IDs...
I, [2016-06-10T00:20:25.678128 #48419]  INFO -- : No API Token set. Using local tests only...
I, [2016-06-10T00:20:25.679177 #48419]  INFO -- :
I, [2016-06-10T00:20:25.679201 #48419]  INFO -- : [VALID]
```

File: spec/exporter_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
   let(:options) do
     instance_double(
       'RainforestCli::Options',
-      token: nil,
+      token: 'token',
```
Comment (juliankrispel) [inline spec/exporter_spec.rb]: This seems a bit hacky. If the token isn't defined, it should just be nil no?

Comment (juliankrispel) [inline spec/exporter_spec.rb]: or undefined...

Comment (juliankrispel) [inline spec/exporter_spec.rb]: oh, whoops, this is a spec, ignore me

**Comment by juliankrispel** [comment]
@epaulet I don't see a regression test to prove that you're getting an exception when the token is not set.  It'd be great to have this for all commands. Other than that lgtm :+1:

**Comment by epaulet** [comment]
You shouldn't get an exception actually. It'll just validate what it can without server data in that case.

**Comment by juliankrispel** [comment]
@epaulet shouldn't it give you a message that you need to specify a token, I had this with all the other commands

File: README.md [hunk 3]
```diff
@@ -62,6 +68,14 @@ You may also specify a custom test title or file name.
 rainforest new "My Awesome Title"
 ```
 
+Validate your tests for syntax and correct RFML ids for embedded tests.
+Use the `--token` options or `RAINFOREST_API_TOKEN` environment variable
+to validate your tests against server data as well.
+
+```bash
+rainforest validate
```
Comment (juliankrispel) [inline README.md]: awesome thanks

**Pull request merged**